### PR TITLE
Refactor swifttool

### PR DESCRIPTION
**Pull request by craigtracey**
This change simply moves SwiftRingsDefinition to its own file. The code
was becoming too unweildy in a single file.

### CODE REVIEW

File: swifttool/ring_defintion.py [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+                    ringdisks = []
+                    # Add all disks designated for ringtype
+                    if isinstance(disks['disks'], dict):
+                        if ringtype in disks['disks']:
+                            ringdisks += disks['disks'][ringtype]
+                        else:
+                            raise Exception("Malformed ring_defintion.yml. "
+                                            "Unknown ringtype: %s" % ringtype)
+                    elif isinstance(disks['disks'], list):
+                        ringdisks = disks['disks']
```
Comment (omgjlk) [inline swifttool/ring_defintion.py:169]: Do we have / need an else block to handle if `disks['disks']` is neither a dict or a list?

**Comment by omgjlk** [comment]
Just one question, I can't see anything obviously wrong with this.

**Comment by ryshah** [comment]
:+1:

File: swifttool/ring_defintion.py (hunk 1 above)
Comment (craigtracey) [inline swifttool/ring_defintion.py:169]: This is not new code...merely moved. I would prefer that we not address any changes to logic in this change. We can revisit in another PR.

**Pull request merged**